src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -88,6 +88,11 @@ public bool IsKeyword(SyntaxToken token)
                 SyntaxFacts.IsKeywordKind(kind); // both contextual and reserved keywords
         }
 
+        public bool IsKeyword(string text)
+        {
+            return SyntaxFacts.GetKeywordKind(text) != SyntaxKind.None; // reserved keywords only
```

Comment:
it is not good that there are two methods named the exact same thing, but they have different behavior.  (specifically, this is commented to be only 'reserved keywords', but the one above says 'contextual and reserved keywords'.

Reply:
Will rename to `IsReservedKeyword`